pandas/core/groupby/generic.py

```diff
@@ -687,7 +689,7 @@ def value_counts(
             # in a backward compatible way
             # GH38672 relates to categorical dtype
             ser = self.apply(
-                Series.value_counts,
+                self._obj_1d_constructor.value_counts,
```

Comment:
this is trouble bc in general we can't assume that _constructor is a class

Reply:
What else could it be?

Reply:
Geopandas has a callable that can dispatch to different classes.  @jorisvandenbossche has argued against deprecating allowing this.

Reply:
I've added a check whether `self._obj_1d_constructor` is a `Series`